source/m2mfirmware.cpp

```diff
@@ -430,5 +430,7 @@ bool M2MFirmware::check_value_range(FirmwareResource resource, int64_t value) co
 
 void M2MFirmware::set_zero_value(M2MResource *resource)
 {
-    resource->set_value((const uint8_t*)"0", 1);
+    if(resource) {
+        resource->set_value((const uint8_t*)"0", 1);
+    }
```

Comment:
Good point on references. In C++ they're possibly the best way to sidestep the whole issue on the language level.

Although it's still possible to pass  `*(resource_t *) NULL` and generate the segmentation fault if you try really hard, and there would be then no way to assert against that.